### DISCUSSION TITLE
How to double element in field array

### DISCUSSION THREAD

**Discussion by PetrovaDaria** [react-hook-form/react-hook-form#6908] [Q&A]
How to append to field array new element with values of one of the existing elements in field array?
I tried next way:
```
const onDouble = useCallback(async (index: number) => {
    await trigger(); // tried with and without this line
    const field = fields[index];
    append(field);
  }, [fields]);
```
but the `field` contains not current values and new field has not correct values

**Comment by lundn** [comment]
Would be very useful if you could provide a codesandbox, because it's hard to know exactly what you want to achieve. Could you check out my codesandbox and see if that meets your criteria? 
https://codesandbox.io/s/react-hook-form-usefieldarray-nested-arrays-forked-2psl5

**Reply by PetrovaDaria** [reply]
I have similar code, but it seems, that when in callback I take field by index, it doesn't have values that I wrote. Values start to appear only after appending/removing/etc (and yes, I have `fields` and `append` dependencies in hook)

**Reply by PetrovaDaria** [reply]
I can't understand why your example is working  because I found another discussion on this topic. It concludes that your example can't work https://github.com/react-hook-form/react-hook-form/discussions/6123

**Reply by PetrovaDaria** [reply]
and your example don't work either. if I change default value, new double field will contain default and not changed value
